core/sail/lmdb/src/main/java/org/eclipse/rdf4j/sail/lmdb/LmdbSailStore.java

```diff
@@ -56,6 +56,89 @@
 
 	private final ValueStore valueStore;
 
+	private final ExecutorService tripleStoreExecutor = Executors.newCachedThreadPool();
+	private final CircularBuffer<Operation> opQueue = new CircularBuffer<>(1024);
+	private volatile Throwable tripleStoreException;
+	private final AtomicBoolean running = new AtomicBoolean(false);
+	private boolean multiThreadingActive;
+	private volatile boolean asyncTransactionFinished;
+	private volatile boolean nextTransactionAsync;
+
+	private boolean enableMultiThreading = true;
```

Comment:
Is this something you want to make configurable?

Reply:
I also thought about making this configurable. When looking at the benchmarks below I am unsure if this is really necessary.

Reply:
Yep, looking at those benchmarks it seems to be an improvement across the board, so I can't immediately see a use case for _not_ wanting multi-threading enabled. Let's leave as-is for now - we can always decide to make it configurable later if we encounter a need for it.